packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -134,6 +155,17 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         return [];
     }
 
+    /**
+     * Check to see if one of the paths is in the current workspace.
+     */
+    protected async waitForItemInWorkspace(): Promise<any> {
+        const doesContain = await this.workspaceService.contains(this.workspaceContains);
```

Comment:
I must be missing something, but why don't we just do
```typescript
protected async waitForItemInWorkspace(): Promise<any> {
  return this.workspaceService.contains(this.workspaceContains);
}
```

Reply:
Just to make sure I got this correctly; `return new Promise(resolve => { });` will not get resolved and that's the difference and the desired behavior here, right? So it will not trigger activation.